leaves - cloudy

```diff
@@ -0,0 +1,120 @@
+########################################################
+# Step 1: Establish the layers
+
+# Which layers are nested in each other?
+##Driver Info is wrapped in a method, which is in a hash, with an array that has hashs of trips. 
+# Which layers of data "have" within it a different layer?
+# Which layers are "next" to each other?
+
+########################################################
+# Step 2: Assign a data structure to each layer
+
+# Copy your list from above, and in this section
+# determine what data structure each layer should have
+
+########################################################
+# Step 3: Make the data structure!
+
+# Setup the entire data structure:
+# based off of the notes you have above, create the
+# and manually write in data presented in rides.csv
+# You should be copying and pasting the literal data
+# into this data structure, such as "DR0004"
+# and "3rd Feb 2016" and "RD0022"
+
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+
+# Use an iteration blocks to print the following answers:
+# - the number of rides each driver has given?
+# The number of rides each driver has given is:
+# Driver DR001: 3 rides
+# Driver DR002: 3 rides
+# Driver DR003: 2 rides
+# Driver DR004: 3 rides
+# - the total amount of money each driver has made?
+# The total earnings per driver is: 
+# Driver DR001: 85 dollars
+# Driver DR002: 75 dollars
+# Driver DR003: 55 dollars
+# Driver DR004: 35 dollars
+# - the average rating for each driver? The average rating per driver is
+# Driver DR001: 3.00 rating
+# Driver DR002: 3.00 rating
+# Driver DR003: 3.50 rating
+# Driver DR004: 4.67 rating
+# - Which driver made the most money? DR001 with: 85 dollars
+# - Which driver has the highest average rating? DR004 with 4.67
+
+
+def driver_info
+  drivers_info = {
```

Comment:
I think I would have liked to see you passing around the `drivers_info` variable, rather than using a method that generates it each time.